cmd/minishift/cmd/addon/addon_apply.go

```diff
@@ -97,3 +100,28 @@ func runApplyAddon(cmd *cobra.Command, args []string) {
 		}
 	}
 }
+
+func determineRoutingSuffix(driver drivers.Driver) string {
+	defer func() {
+		if r := recover(); r != nil {
+			atexit.ExitWithMessage(1, "Unable to determine routing suffix from OpenShift master config.")
+		}
+	}()
+
+	sshCommander := provision.GenericSSHCommander{Driver: driver}
+	dockerCommander := docker.NewVmDockerCommander(sshCommander)
+
+	raw, err := openshift.ViewConfig(openshift.MASTER, dockerCommander)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Unable to retrieve OpenShift master configuration: %s", err.Error()))
+	}
+
+	var config map[interface{}]interface{}
+	err = yaml.Unmarshal([]byte(raw), &config)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Unable to parse OpenShift master configuration: %s", err.Error()))
+	}
+
+	// making assumptions about the master config here. In case the config structure changes, the code might panic here
+	return config["routingConfig"].(map[interface{}]interface{})["subdomain"].(string)
+}
```

Comment:
Not sure what you mean. Are you having potential concerns on the method we retrieve the routing suffix or how I deal with the YAML. In case of the former, I believe this is by far the best way and I really like the we benefit from the infrastructure we have in place.

In case of the latter. There are always two ways to deal with this. I could also makes this a typed un-marshall, but imo this is not required here (and can fail as well, in case the underlying YAML changes). In the worst case the user sees an error message.